src/PrestaShopBundle/Security/OAuth2/Repository/AccessTokenRepository.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Security\OAuth2\Repository;
+
+use League\OAuth2\Server\Entities\AccessTokenEntityInterface;
+use League\OAuth2\Server\Entities\ClientEntityInterface;
+use League\OAuth2\Server\Repositories\AccessTokenRepositoryInterface;
+use PrestaShopBundle\Security\OAuth2\Entity\AccessToken;
+
+class AccessTokenRepository implements AccessTokenRepositoryInterface
```

Comment:
Is this a temporary incomplete implementation? If so and you don't plan on finishing everything here,  maybe it's worth adding some todos to indicate what's left to do?

Reply:
Also, can you describe the purpose of this service exactly:
- generate token is a given
- invalidate token based on a configuration max length?
- persist tokens in a database or somewhere else?
- is it used for all the implementations (including the ones from modules) or is it only for the custom internal PrestaShop implementation?